internal/pkg/composable/providers/kubernetes/pod.go

```diff
@@ -437,8 +437,14 @@ func generateContainerData(
 
 				if config.Hints.Enabled { // This is "hints based autodiscovery flow"
 					if !managed {
-						hintsMapping := getHintsMapping(k8sMapping, logger, config.Prefix, c.ID)
+						hintsMapping, processorMapping := getHintsMapping(k8sMapping, logger, config.Prefix, c.ID)
 						if len(hintsMapping) > 0 {
+							if len(processorMapping) > 0 {
+								for _, processor := range processorMapping {
```

Comment:
This would a need some explanatory comments to explain that the processors are added in the input level etc. Maybe moving this for loop into a standalone helper function along with the comment would be better but it's up to you to decide.

Reply:
Added relevant comment. I had second thoughts and would have been overkill to add a function only for one for

Reply:
I added the function updateProcessors afterall see https://github.com/elastic/elastic-agent/pull/3107/files#diff-ed933ad9cf09eb9ca797c5c89b5e8991624cbe5301d88f996d9bd947f4e3ea16R468